src/journal/journald-kmsg.c

```diff
@@ -385,6 +385,40 @@ static int dispatch_dev_kmsg(sd_event_source *es, int fd, uint32_t revents, void
         return server_read_dev_kmsg(s);
 }
 
+int server_reload_dev_kmsg(Server *s) {
+        int r;
+
+        assert(s);
+
+        mode_t mode = O_CLOEXEC|O_NONBLOCK|O_NOCTTY|
+                (s->read_kmsg ? O_RDWR : O_WRONLY);
+
+        int flags = fcntl(s->dev_kmsg_fd, F_GETFL);
+        if (flags == -1) {
```

Comment:
we generally check for syscall errors via `< 0` not `== -1`, as  amatter of coding style.